packages/terra-form-select/src/single/Frame.jsx

```diff
@@ -184,12 +186,21 @@ class Frame extends React.Component {
   }
 
   getDisplay(displayId, placeholderId) {
-    const { display, placeholder } = this.props;
+    const {
+      children, display, intl, isFilterStyle,
+    } = this.props;
 
-    return (display
-      ? <span id={displayId}>{display}</span>
-      : <div id={placeholderId} className={cx('placeholder')}>{placeholder || '\xa0'}</div>
-    );
+    if (!isFilterStyle && !display) {
+      return (<div id={placeholderId} className={cx('placeholder')}>{intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' }) || '\xa0'}</div>);
+    }
+
+    if (display && MenuUtil.findByDisplay(children, display)) {
+      return (<span id={displayId}>{display}</span>);
+    }
+    if (children) {
```

Comment:
what happens in the case if isFilterStyle is set to false and the display values is not found  ? which fails the above condition.  i guess in that case we add the first item in the display with this logic. which shouldn't be the case as isFilterStyle is set to false.

Reply:
Resolved it here https://github.com/cerner/terra-core/pull/3094/commits/29f1e7283f86fac21d66bd5e739ab2339f154507